### PR TITLE
[BUS-835] add CMS Block and and Element for the Navigation

### PR DESCRIPTION
**Pull request by Marius-BSCM**
add cms block and and element for the sidebar navigation

### CODE REVIEW

File: components/cms/block/CmsBlockCategoryNavigation.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+<template>
+    <CmsLoader
+        :content="content"
+    />
+</template>
+
+<style scoped>
+
+</style>
```
Comment (aheartforspinach) [inline components/cms/block/CmsBlockCategoryNavigation.vue:21]:
```suggestion
```

File: components/cms/block/CmsBlockCategoryNavigation.vue [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const { getSlotContent } = useCmsBlock(props.block);
+
+const content: Schemas['CmsSlot'] = getSlotContent('content');
+</script>
+
+<template>
+    <CmsLoader
+        :content="content"
+    />
```
Comment (aheartforspinach) [inline components/cms/block/CmsBlockCategoryNavigation.vue:16]:
```suggestion
    <CmsLoader :content="content" />
```

File: components/cms/element/CmsElementCategoryNavigation.vue [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+await loadNavigationElements({ depth: 2 });
+
+const isActive = (path: Schemas['SeoUrl'][] | undefined, onlyExactMatch: boolean = false) => {
+    if (!path) return false;
+    const formattedPath = '/' + path[0]?.seoPathInfo;
+    const { path: currentPath } = useRoute();
+    return onlyExactMatch ? formattedPath === currentPath : currentPath.includes(formattedPath);
+}
+</script>
+
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementCategoryNavigation.vue:21]:
```suggestion
```

File: components/cms/element/CmsElementCategoryNavigation.vue [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+const { navigationElements, loadNavigationElements } = useNavigation();
+
+await loadNavigationElements({ depth: 2 });
+
+const isActive = (path: Schemas['SeoUrl'][] | undefined, onlyExactMatch: boolean = false) => {
+    if (!path) return false;
+    const formattedPath = '/' + path[0]?.seoPathInfo;
+    const { path: currentPath } = useRoute();
+    return onlyExactMatch ? formattedPath === currentPath : currentPath.includes(formattedPath);
+}
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementCategoryNavigation.vue:19]: Could you move this code to a composable? We used the same code in LayoutHeader.vue Maybe you could use the composable also there

File: components/cms/element/CmsElementCategoryNavigation.vue [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                    </NuxtLink>
+                </li>
+            </ul>
+        </li>
+    </ul>
+</template>
+
+
+<style scoped>
+</style>
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementCategoryNavigation.vue:51]:
```suggestion
```

File: components/cms/element/CmsElementCategoryNavigation.vue [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    if (!path) return false;
+    const formattedPath = '/' + path[0]?.seoPathInfo;
+    const { path: currentPath } = useRoute();
+    return onlyExactMatch ? formattedPath === currentPath : currentPath.includes(formattedPath);
+}
+</script>
+
+
+<template>
+    <ul v-if="navigationElements && navigationElements.length">
```
Comment (aheartforspinach) [inline components/cms/element/CmsElementCategoryNavigation.vue:24]:
```suggestion
    <ul v-if="navigationElements?.length">
```

File: components/cms/element/CmsElementCategoryNavigation.vue (hunk 4 above)
Comment (Marius-BSCM) [reply components/cms/element/CmsElementCategoryNavigation.vue]: I outsourced the isActive logic into a composable. I then used the composable in this element and in the NavigationLink.vue. Hope thats fine this way :)

**Review by aheartforspinach — APPROVED** [review]

File: components/cms/element/CmsElementCategoryNavigation.vue (hunk 4 above)
Comment (aheartforspinach) [reply components/cms/element/CmsElementCategoryNavigation.vue]: thank you!

**Pull request merged**